Sources/TidepoolKit/Auth/OAuth2Authenticator.swift

```diff
@@ -9,19 +9,53 @@
 import Foundation
 import AuthenticationServices
 
+public protocol OAuth2AuthenticatorSessionProvider {
+    func startSession(authURL: URL, callbackScheme: String?) async throws -> URL
+}
+
+@MainActor
+public class ASWebAuthenticationSessionProvider: OAuth2AuthenticatorSessionProvider {
+
+    private let contextProviding: ASWebAuthenticationPresentationContextProviding
+    private var authenticationSession: ASWebAuthenticationSession?
+
+    public init(contextProviding: ASWebAuthenticationPresentationContextProviding) {
+        self.contextProviding = contextProviding
+    }
+
+    public func startSession(authURL: URL, callbackScheme: String?) async throws -> URL {
+        let callbackURL: URL = try await withCheckedThrowingContinuation { continuation in
+            self.authenticationSession = ASWebAuthenticationSession(url: authURL, callbackURLScheme: callbackScheme) { callbackURL, error in
+                if let error {
+                    continuation.resume(throwing: error)
+                    return
+                }
+
+                guard let callbackURL = callbackURL else {
+                    continuation.resume(throwing: TError.missingAuthenticationState)
+                    return
+                }
+
+                continuation.resume(returning: callbackURL)
+            }
+            authenticationSession?.presentationContextProvider = contextProviding
+            authenticationSession?.start()
+        }
+        return callbackURL
```

Comment:
Did you need to declare the variable and type here to satisfy the compiler? I know sometimes it's finicky with the `continuations`. I think you might be able to get around that with this:
```swift
public func startSession(authURL: URL, callbackScheme: String?) async throws -> URL {
    try await withCheckedThrowingContinuation<URL> { continuation in
        ...
    }
}
```

Reply:
Ah, yeah, thanks. Sometimes the compiler will throw a nonsensical error, and adding explicit types helps it throw a better error.  I fixed the real issue and forgot to cleanup the explicit types. Pushed a change to clean this up.